hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -296,7 +296,7 @@ public MessageType convertAvroSchemaToParquet(Schema schema) {
   public static boolean isSchemaCompatible(Schema oldSchema, Schema newSchema) {
     if (oldSchema.getType() == newSchema.getType() && newSchema.getType() == Schema.Type.RECORD) {
       // record names must match:
-      if (!SchemaCompatibility.schemaNameEquals(oldSchema, newSchema)) {
+      if (!SchemaCompatibility.schemaNameEquals(newSchema, oldSchema)) {
```

Comment:
Hi @nsivabalan @prashantwason ,I have test write a "int" type to "long" type.It fails in the the schema validate .
However,I write a "long" to "int", it can pass the validate. So I think this reverse here introduce some problem.